tests/test_io.py

```diff
@@ -1405,3 +1405,94 @@ def test_read_sql_dict_str_value_dtype() -> None:
             DataFrame,
         )
         con.close()
+
+
+def test_added_date_format() -> None:
+    with ensure_clean() as path:
+        df_dates = pd.DataFrame(
+            data={
+                "col1": ["2023-03-15", "2023-04-20"],
+            }
+        )
+        df_dates.to_csv(path)
+
+        check(
+            assert_type(
+                pd.read_csv(path, parse_dates=["col1"], date_format="%Y-%m-%d"),
+                pd.DataFrame,
+            ),
+            pd.DataFrame,
+        )
+        check(
+            assert_type(
+                pd.read_csv(
+                    path, parse_dates=["col1"], date_format={"col1": "%Y-%m-%d"}
+                ),
+                pd.DataFrame,
+            ),
+            pd.DataFrame,
+        )
+        check(
+            assert_type(
+                pd.read_csv(path, parse_dates=["col1"], date_format={10: "%Y-%m-%d"}),
```

Comment:
```suggestion
                pd.read_csv(path, parse_dates=["col1"], date_format={0: "%Y-%m-%d"}),
```

Use 0 in the dict, not 10, and do this in the other tests as well